### PR TITLE
Api 980 - Datetimepicker input

### PR DESCRIPTION
**Pull request by jocteau**
Ignore my commits, but I need to build and push to the branch to test every time I do a modification

## Jira Ticket 
[https://citizennet.atlassian.net/browse/API-980](https://citizennet.atlassian.net/browse/API-980)

## Issue
From my understanding we are triggering the flexform property `watch` whenever `ngModel` change. 
For the `cn-datetimepicker` that would get triggered every time the user would input a character into the input field:
```
                    "watch": {
                        "condition": "model.startDate > model.stopDate",
                        "resolution": "model.stopDate = model.startDate + 1 hours",
                    },
```
Which makes impossible to enter a date through typing for the `stopDate`

## More
@davezuch I don't know if it's the best solution, maybe there is a more "straight forward" way of actualizing the `ngModel` only when the dropdown is closed 🤷🏻‍♂️

### CODE REVIEW

**Review by matt-morales — APPROVED** [review]
Nice, this looks good to me. I'll defer to @davezuch for any recommendations since I am not as familiar with this code

File: src/cn-datetimepicker.js [hunk 1]
```diff
@@ -38,14 +38,14 @@
       restrict: 'E',
       require: 'ngModel',
       template: `
-          <div dropdown class="dropdown">
+          <div dropdown class="dropdown" on-toggle="dopdownToggled(open)">
```
Comment (davezuch) [inline src/cn-datetimepicker.js]:
```suggestion
          <div dropdown class="dropdown" on-toggle="dropdownToggled(open)">
```

File: src/cn-datetimepicker.js [hunk 2]
```diff
@@ -114,27 +115,47 @@
...
         if($scope.required) {
-          ctrl.$setValidity('tv4-302', !!($scope.ngModel || $scope.ngModel === 0));
+          ctrl.$setValidity('tv4-302', !!($scope.localNgModel || $scope.localNgModel === 0));
         }
-        if(!angular.equals(newVal, prevVal)) {
-          ctrl.$setDirty();
+        $scope.ngModel = $scope.localNgModel;
+      };
+
+      $scope.dopdownToggled = (open) => {
```
Comment (davezuch) [inline src/cn-datetimepicker.js]:
```suggestion
      $scope.dropdownToggled = (open) => {
```

**Review by davezuch — APPROVED** [review]
> Ignore my commits, but I need to build and push to the branch to test every time I do a modification

You should be able to use [`bower link`](https://bower.io/docs/api/#link) or just manually symlink, to avoid constantly committing and pushing changes.

> @davezuch I don't know if it's the best solution, maybe there is a more "straight forward" way of actualizing the `ngModel` only when the dropdown is closed 🤷🏻‍♂️

I think [`ngModelOptions`](https://docs.angularjs.org/api/ng/directive/ngModelOptions) might be what you want. Seems like setting `updateOn: 'blur'` might offer a more straight-forward solution.

Assuming this works (I haven't tested), I don't want to block in case it's urgent, but if not I recommend trying the `ngModelOptions` solution.

**Comment by jocteau** [comment]
> You should be able to use [`bower link`](https://bower.io/docs/api/#link) or just manually symlink, to avoid constantly committing and pushing changes.
> 
Wow, life savior. Thanks!


> I think [`ngModelOptions`](https://docs.angularjs.org/api/ng/directive/ngModelOptions) might be what you want. Seems like setting `updateOn: 'blur'` might offer a more straight-forward solution.

I feel dumb now 🤦‍♂️ 

I updated my PR, it looks way cleaner now...

File: src/cn-datetimepicker.js [hunk 3]
```diff
@@ -113,7 +114,7 @@
       };
 
       //////////
-
+
```
Comment (davezuch) [inline src/cn-datetimepicker.js]:
```suggestion
 
```

**Review by davezuch — APPROVED** [review]
> Wow, life savior. Thanks!

Glad to help!

> I feel dumb now 🤦‍♂️

Haha, no worries, not like you would just know to look for that. I'm pretty sure I've done the same thing before.

> I updated my PR, it looks way cleaner now...

Looks great!

**Pull request merged**